java-manta-client/pom.xml

```diff
@@ -170,6 +170,20 @@
             <version>${dependency.urlbuilder.version}</version>
         </dependency>
 
+        <!-- object cloning library for memoization of encryption state -->
```

Comment:
There is a version incompatibility between the newest release of `kryo` (which depends on 2.1) and the available versions of `uk.com.robust-it:cloning`:

kryo-4.0.0 depends on objenesis-2.1
kryo-master depends on objenesis-2.5.1 (though it is uncertain when the next release of kryo will occur)

cloning-1.9.5 depends on objenesis-2.5.1
cloning-1.9.4 depends on objenesis-2.2

The [release notes](https://objenesis.org/notes.html) for objenisis don't seem to imply any breaking changes between 2.1 and 2.5.1 so the issue is specifically related to version numbers.

I used a separate sandbox project depending on java-manta 3.1.4-SNAPSHOT to check the impact of not shading and am confused by the resulting dependency tree which doesn't mention objenesis at all:

```
mvn clean dependency:tree -Dverbose
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building j 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ j ---
[INFO]
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ j ---
[INFO] co.tjcelaya.sandbox:j:jar:1.0-SNAPSHOT
[INFO] +- com.joyent.manta:java-manta-client:jar:3.1.4-SNAPSHOT:compile
[INFO] |  +- org.bouncycastle:bcpkix-jdk15on:jar:1.56:compile
[INFO] |  |  \- (org.bouncycastle:bcprov-jdk15on:jar:1.56:compile - omitted for duplicate)
[INFO] |  +- org.bouncycastle:bcprov-jdk15on:jar:1.56:compile
[INFO] |  +- com.squareup.jnagmp:bouncycastle-rsa:jar:2.0.0:compile
[INFO] |  |  \- com.squareup.jnagmp:jnagmp:jar:2.0.0:compile
[INFO] |  |     \- net.java.dev.jna:jna:jar:4.0.0:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.21:compile
[INFO] +- com.joyent.manta:java-manta-client-kryo-serialization:jar:3.1.4-SNAPSHOT:compile
[INFO] |  +- (com.joyent.manta:java-manta-client:jar:3.1.4-SNAPSHOT:compile - omitted for duplicate)
[INFO] |  +- com.esotericsoftware:kryo:jar:4.0.0:compile
[INFO] |  |  \- com.esotericsoftware:reflectasm:jar:1.11.3:compile
[INFO] |  |     \- org.ow2.asm:asm:jar:5.0.4:compile
[INFO] |  +- com.dorkbox:MinLog-SLF4J:jar:1.8:compile
[INFO] |  \- (org.slf4j:slf4j-api:jar:1.7.21:compile - omitted for duplicate)
[INFO] +- org.slf4j:slf4j-simple:jar:1.7.25:compile
[INFO] |  \- (org.slf4j:slf4j-api:jar:1.7.25:compile - omitted for conflict with 1.7.21)
[INFO] +- commons-cli:commons-cli:jar:LATEST:compile
[INFO] +- commons-io:commons-io:jar:2.5:compile
[INFO] \- org.apache.commons:commons-lang3:jar:3.4:compile```